ncm-opennebula/src/main/perl/OpenNebula/Network.pm

```diff
@@ -0,0 +1,208 @@
+#${PMpre} NCM::Component::OpenNebula::Network${PMpost}
+
+=head1 NAME
+
+NCM::Component::OpenNebula::Network adds C<OpenNebula> C<VirtualNetwork> 
+configuration support to L<NCM::Component::OpenNebula>.
+
+=head2 Public methods
+
+=over
+
+
+=item update_vn_ar
+
+Updates C<VirtualNetwork> C<ARs>.
+
+=cut
+
+sub update_vn_ar
+{
+    my ($self, $one, $vnetname, $template, $t, $data) = @_;
+    my $arid;
+
+    my %ar_opts = ('template' => $template);
+    $arid = $t->get_ar_id(%ar_opts);
+    $self->debug(1, "Detected AR id to update: ", $arid);
+    if (defined($arid)) {
+        $data->{$vnetname}->{ar}->{ar_id} = "$arid";
+        $template = $self->process_template($data, "vnet");
```

Comment:
you are redefining the `$tempalte` that was passed as argumnet. can you use a new variable for this to distinguish this